engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/result/NadelResultTransformer.kt

```diff
@@ -31,32 +35,85 @@ internal class NadelResultTransformer(private val executionBlueprint: NadelOvera
         service: Service,
         result: ServiceExecutionResult,
     ): ServiceExecutionResult {
-        val instructions = executionPlan.transformationSteps.flatMap { (field, steps) ->
-            steps.flatMap step@{ step ->
+        val nodes = JsonNodes(result.data, executionContext.hints)
+
+        suspend fun getAsync(): List<NadelResultInstruction> {
+            val deferredInstructions = ArrayList<Deferred<List<NadelResultInstruction>>>()
+
+            coroutineScope {
+                for ((field, steps) in executionPlan.transformationSteps) {
+                    // This can be null if we did not end up sending the field e.g. for hydration
+                    val underlyingFields = overallToUnderlyingFields[field]
+                    if (underlyingFields == null || underlyingFields.isEmpty()) {
+                        continue
+                    }
+
+                    for (step in steps) {
+                        deferredInstructions.add(
```

Comment:
Just to note, no need to synchronize the list here since it's sequential. It's only parallel inside the async block.